platform/posix/ota_pal/source/ota_pal_posix.c

```diff
@@ -433,17 +434,35 @@ OtaPalStatus_t otaPal_CreateFileForRx( OtaFileContext_t * const C )
             {
                 /* POSIX port using standard library */
                 /* coverity[misra_c_2012_rule_21_6_violation] */
-                C->pFile = fopen( ( const char * ) realFilePath, "w+b" );
-
                 if( C->pFile != NULL )
                 {
-                    result = OTA_PAL_COMBINE_ERR( OtaPalSuccess, 0 );
-                    LogInfo( ( "Receive file created." ) );
+                    filerc = fclose( C->pFile );
+
+                    if( filerc != 0 )
+                    {
+                        LogError( ( "Failed to close OTA update file." ) );
+                        result = OTA_PAL_COMBINE_ERR( OtaPalFileClose, errno );
+                    }
+                    else
```

Comment:
(Nit) `filerc` can be avoided
```suggestion
                    if( fclose( C->pFile ) != 0)
                    {
                        LogError( ( "Failed to close OTA update file." ) );
                        result = OTA_PAL_COMBINE_ERR( OtaPalFileClose, errno );
                    }
                    else
```